3rdparty/python/requirements.txt

```diff
@@ -47,3 +47,6 @@ uvicorn[standard]==0.17.6
 
 beautifulsoup4==4.11.1 # Util for webcrawling for pulling in tool versions
 python-gnupg==0.4.9 # For validating signatures
+
+# Only used for release management
+PyGithub==2.0.0rc1
```

Comment:
We can enforce this, by adding `("[PyGithub]", "src/python/pants_release/**",),` to
https://github.com/pantsbuild/pants/blob/00786d3a8d35bf82db128756a5b821f08e38cf53/3rdparty/python/BUILD#L36-L49

Something like:
```python
__dependents_rules__(
    (  # Only the explorer server may depend on these libraries
        (
            "[fastapi]",
            "[starlette]",
            "[strawberry-graphql]",
            "[uvicorn]",
        ),
        "src/python/pants/explorer/server/**",
        "!*",
    ),

    ("[PyGithub]", "src/python/pants_release/**",),

    # Free for all on the rest
    ("*", "*"),
)
```

Reply:
As a slight aside, should we swap so the core pants package is includelisted, and everything _else_ is free game?
Because it shouldn't matter if a release needs `fastapi` (I mean why, but you get the point) or some non-included plugin uses `PyGitHub` (what if its a GitHub plugin?)

Reply:
I've filed https://github.com/pantsbuild/pants/issues/19458

Reply:
> As a slight aside, should we swap so the core pants package is includelisted, and everything _else_ is free game? Because it shouldn't matter if a release needs `fastapi` (I mean why, but you get the point) or some non-included plugin uses `PyGitHub` (what if its a GitHub plugin?)

That works.